### ISSUE TITLE
Popover with select

### ISSUE THREAD

**Issue by Deepak-Kesavan** [ant-design/ant-design#41629]
### Reproduction link

[![Edit on CodeSandbox](https://codesandbox.io/static/img/play-codesandbox.svg)](https://codesandbox.io/s/popover-with-select-in-ant-5-x-0kpm3x)

### Steps to reproduce

Dynamically render a select component inside popover based on a state value and add a remove button to remove the rendered select.

Open the popover and change a value in select dropdown, then remove the rendered select by pressing the close button. It will throw an error.

### What is expected?

It is expected to work without errors like it is working here in antd 4.x. https://codesandbox.io/s/popover-with-select-in-ant-4-x-4j3eup

### What is actually happening?

It is throwing an error `Exception has occurred: TypeError: Cannot read properties of null (reading 'contains')` from node_modules `node_modules/@rc-component/trigger/es/index.js:127`.

| Environment | Info |
| --- | --- |
| antd | 5.3.2 |
| React | 18.x (18.2.0) |
| System | Pop!_OS 22.04 LTS (Linux) |
| Browser | Brave |

---
It is working fine in ant 4.x and breaks in 5.x

**Comment by MadCcc** [comment]
Try to remove `node_modules` and lock file and reinstall. This should be fixed.

**Comment by zombieJ** [comment]
Reinstall `node_modules`, this is already fixed : )

**Comment by Deepak-Kesavan** [comment]
> Try to remove `node_modules` and lock file and reinstall. This should be fixed.

I did everything already more than 10 - 20 times and with different versions. I am trying to fix this over a week. Deleted `node_modules` and `package-lock.json` then reinstalled. Nothing worked. Check the reproduction link. I created it just now and it throws the same error.

**Comment by Deepak-Kesavan** [comment]
> Reinstall `node_modules`, this is already fixed : )

@zombieJ If it is fixed, can you point out the version in which it is fixed? I am having the current version `5.4.0` in my local which is a newly created project and it gives the same error.

**Comment by zombieJ** [comment]
`@rc-component/trigger@1.6.3`

**Comment by Deepak-Kesavan** [comment]
> `@rc-component/trigger@1.6.3`

@zombieJ Thanks. Looks like it was fixed yesterday in `5.4.0`. But there is one more issue that needs to be addressed. 

When removing a select component with x button, it also closes the popover which I believe is unintended. 

For your reference:

[ant 4.x](https://codesandbox.io/s/popover-with-select-in-ant-4-x-4j3eup) - works fine
[ant 5.4.0](https://codesandbox.io/s/popover-with-select-in-ant-5-x-0kpm3x) - popover closes when I press the x button.

Anyway for now I handled it with `event.stopPropagation();`.
